### PR TITLE
KT-11990 J2K `native` to `external` conversion

### PR DESCRIPTION
**Pull request by semoro**
https://youtrack.jetbrains.com/issue/KT-11990

### CODE REVIEW

File: j2k/testData/fileOrElement/function/nativeMethods.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+public class Foo {
+    private native final void nativeMethod()
```
Comment (yole) [inline j2k/testData/fileOrElement/function/nativeMethods.java]: Please avoid irrelevant syntax errors in testdata. Java does require semicolons :)

File: j2k/src/org/jetbrains/kotlin/j2k/ClassBodyConverter.kt [hunk 2]
```diff
@@ -152,13 +152,14 @@ class ClassBodyConverter(private val psiClass: PsiClass,
...
         when (member) {
             is PsiMethod -> {
-                memberToPropertyInfo[member]?.let { propertyInfo ->
-                    if (propertyInfo.field != null) return null // just drop the method, property will be generated when converting the field
-                    return if (member == propertyInfo.getMethod || propertyInfo.getMethod == null)
-                        converter.convertProperty(propertyInfo, classKind)
-                    else
-                        null // drop the method, property will be generated when converting the field or the getter
-                }
+                if (!member.hasModifierProperty(PsiModifier.NATIVE))
```
Comment (yole) [inline j2k/src/org/jetbrains/kotlin/j2k/ClassBodyConverter.kt]: This doesn't seem to be a correct check. This is valid Kotlin:

``` kotlin
val foo: Int = 0
    external get
```

**Comment by semoro** [comment]
Why we must initialize val with external get?

**Comment by yole** [comment]
I think it's a bug in the validator; there is no other reason

**Comment by yole** [comment]
Thanks for your contribution!

**Pull request merged**